docs/source/advanced.rst

```diff
@@ -178,26 +178,126 @@ This is basically the same as calling :py:func:`~scikeras.wrappers.BaseWrapper.g
 Data Transformers
 ^^^^^^^^^^^^^^^^^
 
-In some cases, the input actually consists of multiple inputs. E.g.,
+Keras supports a much wider range of inputs/outputs than Scikit-Learn does. E.g.,
 in a text classification task, you might have an array that contains
 the integers representing the tokens for each sample, and another
-array containing the number of tokens of each sample. SciKeras has you
-covered here as well.
+array containing the number of tokens of each sample.
+
+In order to reconcile Keras' expanded input/output support and Scikit-Learn's more
+limited options, SciKeras introduces "data transformers". These are really just
+dependency injection points where you can declare custom data transformations,
+for example to split an array into a list of arrays, join ``X`` & ``y`` into a ``Dataset``, etc.
+In order to keep these transformations in a familiar format, they are implemented as
+sklearn-style transformers. You can think of this setup as an sklearn Pipeline:
+
+.. code-block::
+
+                                   ↗ feature_encoder ↘
+    SciKeras.fit(features, labels)                    dataset_transformer → keras.Model.fit(data)
+                                   ↘ target_encoder  ↗ 
+
+
+Within SciKeras, this is roughly implemented as follows:
+
+.. code:: python
```

Comment:
I think a lot of this code block is unnecessary. I'd use this:

``` python
class PseudoBaseWrapper:
    def fit(self, X, y, sample_weight):
        self.target_encoder_ = self.target_encoder.fit(y)
        self.feature_encoder_ = self.feature_encoder.fit(X)
        self.model_ = self._build_keras_model()
        X, y, sample_weight = self.dataset_encoder.fit(
            self.feature_encoder_.transform(X), self.target_encoder_.transform(y), sample_weight
        )
        self.model_.fit(X=X, y=y, sample_weight=sample_weight)  # tf.keras.Model.fit
        return self

    def predict(self, X):
        X = self.feature_encoder_.transform(X)
        X, _, _ = self.dataset_transformer_.fit_transform((X, None, None))
        y_pred = self.model_.predict(X)
        return self.target_encoder_.inverse_transform(y_pred)
```

Reply:
I agree that we can remove the `@property` stuff. I'll implement that.